File: 1.合并区间/cpp/56.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> segments;
        for (auto& interval: intervals) {
            if (segments.size() == 0 
                || segments.back().back() < interval[0]) {
                segments.push_back(interval);
            } else if (segments.back().back() < interval[1]){
                segments.back().back() = interval[1];
            }
        }
        return segments;  
    }
};

File: 1.合并区间/go/56.go
import "sort"

//author : baronliu

func merge(intervals [][]int) [][]int {
	var result [][]int

	if len(intervals) < 1 {
		return result
	}

	sort.Slice(intervals, func(i int, j int) bool {
		return intervals[i][0] < intervals[j][0]
	})

	for i := 0; i < len(intervals); i++ {
		temp := intervals[i]
		for j := i + 1; j < len(intervals); j++ {
			if intervals[j][0] > temp[1] {
				break
			} else if intervals[j][1] > temp[1] {
				temp[1] = intervals[j][1]
			}
			i++
		}

		result = append(result, temp)
	}
	return result
}